ts/components/MemberListItem.tsx

```diff
@@ -161,43 +157,61 @@ const StyledGroupStatusText = styled.span<{ isFailure: boolean }>`
   text-align: start;
 `;
 
+function localisedStatusFromMemberStatus(memberStatus: MemberStateGroupV2) {
+  switch (memberStatus) {
+    case 'INVITE_FAILED':
+      return window.i18n('groupInviteFailed');
+    case 'INVITE_NOT_SENT':
+      return window.i18n('groupInviteNotSent');
+    case 'INVITE_SENDING':
+      return window.i18n('groupInviteSending', { count: 1 });
+    case 'INVITE_SENT':
+      return window.i18n('groupInviteSent');
+    case 'INVITE_UNKNOWN': // fallback, hopefully won't happen in production
+      return window.i18n('groupInviteStatusUnknown');
+    case 'PROMOTION_UNKNOWN': // fallback, hopefully won't happen in production
+      return window.i18n('adminPromotionStatusUnknown');
+    case 'REMOVED_UNKNOWN': // fallback, hopefully won't happen in production
+    case 'REMOVED_MEMBER': // we want pending removal members at the end of the "invite" states
+    case 'REMOVED_MEMBER_AND_MESSAGES':
+      return null; // no text for those 3 pending removal states
+    case 'PROMOTION_FAILED':
+      return window.i18n('adminPromotionFailed');
+    case 'PROMOTION_NOT_SENT':
+      return window.i18n('adminPromotionNotSent');
+    case 'PROMOTION_SENDING':
+      return window.i18n('adminSendingPromotion', { count: 1 });
+    case 'PROMOTION_SENT':
+      return window.i18n('adminPromotionSent');
+    case 'PROMOTION_ACCEPTED':
+      return null; // no statuses for accepted state;
+    case 'INVITE_ACCEPTED':
+      return null; // no statuses for accepted state
+    default:
+      assertUnreachable(memberStatus, 'Unhandled switch case');
+      return Number.MAX_SAFE_INTEGER;
+  }
+}
+
 const GroupStatusText = ({ groupPk, pubkey }: { pubkey: PubkeyType; groupPk: GroupPubkeyType }) => {
-  const groupInviteFailed = useMemberInviteFailed(pubkey, groupPk);
-  const groupPromotionFailed = useMemberPromotionFailed(pubkey, groupPk);
-  const groupPromotionSending = useMemberPromoteSending(groupPk, pubkey);
+  const memberStatus = useMemberStatus(pubkey, groupPk);
 
-  const groupInviteSent = useMemberInviteSent(pubkey, groupPk);
-  const groupPromotionSent = useMemberPromotionSent(pubkey, groupPk);
-  const groupInviteSending = useMemberInviteSending(groupPk, pubkey);
+  if (!memberStatus) {
+    return null;
+  }
 
   /**
    * Note: Keep the "sending" checks here first, as we might be "sending" when we've previously failed.
```

Comment:
Should this docstring comment not be moved into `localisedStatusFromMemberStatus`?

Reply:
removed, this comment is not relevant anymore